src/ndpp_ops.F90

```diff
@@ -0,0 +1,1083 @@
+      end if
+      ! Get the energy bins (not checking, will assume from here on out we have
+      ! the right data set), and the other meta information
+      allocate(energy_bins(NG + 1))
+      read(UNIT=in) energy_bins, scatt_type, lib_order, &
+        nuscatter, chi_present, mu_bins, thin_tol
+
+      ! set lib_order to the right number for allocating the outgoing array
+      if (scatt_type == SCATT_TYPE_LEGENDRE) lib_order = lib_order + 1
+
+      ! Get \sigma_{s,g'->g,l}(E_{in}) data
+      ! Get elastic Ein information
+
+      ! First find our data length, doesnt depend on nuc or sab
+      read(UNIT=in) NEin
+
+      ! Now allocate all that will be filled
+      allocate(this % el_Ein(NEin))
+      allocate(this % el_Ein_srch(NG + 1))
+      allocate(this % el(NEin))
+
+      ! Now read in el_Ein and el_Ein_srch
+      read(UNIT=in) this % el_Ein
+      read(UNIT=in) this % el_Ein_srch
+
+      ! Now we can get the elastic moments themselves one-by-one due to sparse
+      ! storage nature.
+      do iE = 1, NEin
+        ! get gmin and gmax
+        read(UNIT=in) gmin, gmax
+
+        if ((gmin > 0) .and. (gmax > 0)) then
+          ! Then we can allocate the space. Do it to lib_order
+          ! since this is the largest order requested in the tallies.
+          ! Since we only need to store up to the maximum, we also need to have
+          ! an array for reading the file which we can later truncate to fit
+          ! in to this % el(iE) % outgoing.
+          allocate(temp_outgoing(lib_order, gmin : gmax))
+
+          ! Now we have a space to store the data, get it.
+          read(UNIT=in) temp_outgoing
+          ! And copy in to this % el
+          allocate(this % el(iE) % outgoing(scatt_order, gmin : gmax))
+          this % el(iE) % outgoing(:, gmin : gmax) = &
+            temp_outgoing(1 : scatt_order, gmin : gmax)
+          deallocate(temp_outgoing)
+        end if
+      end do
+
+      ! The remainder only apply to nuclides (inelastic, nu-inelastic and chi data)
+      if (is_nuc) then
+        read(UNIT=in) NEin
+
+        if (NEin > 0) then
+          ! Now allocate all that will be filled
+          allocate(this % inel_Ein(NEin))
+          allocate(this % inel_Ein_srch(NG + 1))
+          allocate(this % inel(NEin))
+
+          ! Now read in inel_Ein, inel_Ein_srch
+          read(UNIT=in) this % inel_Ein
+          read(UNIT=in) this % inel_Ein_srch
+
+          ! Get the inelastic moments themselves
+          do iE = 1, NEin
+            ! get gmin and gmax
+            read(UNIT=in) gmin, gmax
+
+            if ((gmin > 0) .and. (gmax > 0)) then
+              ! Then we can allocate the space. Do it to lib_order
+              ! since this is the largest order requested in the tallies.
+              ! Since we only need to store up to the maximum, we also need to have
+              ! an array for reading the file which we can later truncate to fit
+              ! in to this % inel(iE) % outgoing.
```

Comment:
80 columns.  Gosh, we really should try to get our CI to check these pedantic little formatting things.  My UROP project senses are tingling.